### ISSUE TITLE
Tooltips not working anymore?

### ISSUE THREAD

**Issue by kalekje** [sumatrapdfreader/sumatrapdf#2391]
On an older version of Sumatra, I had no issues with tooltips (plain text). With the new version (or with beta), I am not seeing them.

See the last page of this document for examples: https://muug.ca/mirror/ctan/macros/latex/contrib/pdfcomment/doc/example.pdf

**Comment by GitHubRulesOK** [comment]
There are a few annotations in that sample that are not "normal" for example the highlight that straddles two pages since annotation should be confined to one page at a time, although hyperlinks will be in one pointing to another page.

 However the latex use of tooltips is another offbeat application and reported already in https://github.com/sumatrapdfreader/sumatrapdf/issues/2083 thus closing as a duplicate extension of that issue

**Comment by GitHubRulesOK** [comment]
I dont see an issue in pre-release with normal annotation tooltips that are listed as standard annotations
![image](https://user-images.githubusercontent.com/12754913/229216633-cddf3cb5-6090-479b-9565-9cb41f266ef9.png)

The issue here are the latex less common ones which are NOT listed in Tracker as annotations.
![image](https://user-images.githubusercontent.com/12754913/229217697-593727fc-80f1-4fc5-94f0-4276c86a4024.png)
![image](https://user-images.githubusercontent.com/12754913/229217957-c6c3bd0c-8ab0-4f33-95b4-044a653554ef.png)
![image](https://user-images.githubusercontent.com/12754913/229218410-822e174c-2dc2-4295-9b13-ed83846a3202.png)

**Comment by tobidelbruck** [comment]
Yes, 3.4.6 still has broken tooltip popups. See e.g. https://www.overleaf.com/read/fyvbnxpwwjss where the overleaf PDF built in viewer shows them, and so does Adobe Acrobat Reader and Acrobat DC.
You need to revert back to SumatraPDF 3.0 to get them to work again.
Apparently it is a regression in the underlying PDF engine that Sumatra uses that I could not figure out how to patch with my limited coding skills. See https://github.com/sumatrapdfreader/sumatrapdf/issues/2083 for details.

**Comment by GitHubRulesOK** [comment]
Overleaf has them defined by repetition
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/30c42c54-e4e8-4ee0-8ff2-88087ad03705)
we can see there are 8 of those
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/d3963876-cf99-49c3-8017-d8a586464625)

Looking at PDF we see there are 8 entries of this format

8 0 obj
<</BS<</W 0>>/C[]/F 768/FT/Btn/Ff 65536/H/N/Rect[410.476 654.247 427.22 662.416]/Subtype/Widget/T(tooltip zref@0)/TU(Optical Flow)/Type/Annot/AP<</N 28 0 R/D 29 0 R>>>>
endobj

the point to note is these are tagged `/Btn` widget which MuPDF seems to have stated they did but do not / wont support as per https://github.com/sumatrapdfreader/sumatrapdf/issues/1914#issuecomment-897049428

**Comment by tobidelbruck** [comment]
These comments were extremely helpful, indeed the source of the problem is not sumatrapdf or PDF.js but rather the tip itself created by latex [pdfcomment.](https://ctan.org/tex-archive/macros/latex/contrib/pdfcomment) 

Gemini AI helped me create a patch to pdfcomment. See this [acronyms_with_tooltips](https://www.overleaf.com/read/fyvbnxpwwjss#6de35b) for a working overleaf project for Acrobat and Sumatra that replaces the button with a rectangle annotation. (It does not render correctly in overleaf, but does in the others).

**Comment by GitHubRulesOK** [comment]
For reference the Annotation is seen in 

[acronyms_with_tooltip_test.pdf](https://github.com/user-attachments/files/28670586/acronyms_with_tooltip_test.pdf)  

as a "square" (rectangle) and thus in 3.7etc. its content shown in a tooltip

<img width="725" height="188" alt="Image" src="https://github.com/user-attachments/assets/9bbb2a65-2110-40c1-88b7-88f504d038bd" />
